### PR TITLE
more semantic tests

### PR DESCRIPTION
**Pull request by moben**

### CODE REVIEW

File: tests/semantic/compare_different_classes.invalid.mj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class C1 {}
+class C2 {}
+
+class Test {
+  public static void main(String[] args) {
+    new Test().test();
+  }
+
+  public int test() {
+    if (new C1() == new C2()) {
```
Comment (parttimenerd) [inline tests/semantic/compare_different_classes.invalid.mj]: This is valid (Mini)Java, from the Java 8 Language Specification:

> The equality operators may be used to compare two operands that are convertible
> (§5.1.8) to numeric type, or two operands of type boolean or Boolean, or two operands that are each of either reference type or the null type. All other cases result in a compile-time error.

File: tests/semantic/invalid_lvalue2.invalid.mj [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    true = true;
+
+    1 = 3 + 2;
+
+    null = new Foo();
+
+    Foo f;
+
+    f.f() = 10;
+  }
```
Comment (parttimenerd) [inline tests/semantic/invalid_lvalue2.invalid.mj]: Please split this test case.

File: tests/semantic/undefined_class_type_usage.invalid.mj [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Test {
+  public static void main(String[] args) {}
+
+  public SomeType someType;
+  public SomeType[] someTypeArray;
+
+  public void test () {
+    SomeType someType;
+    SomeType[] someTypeArray;
+  }
```
Comment (parttimenerd) [inline tests/semantic/undefined_class_type_usage.invalid.mj]: Maybe split this test case too.

File: tests/semantic/compare_different_classes.invalid.mj (hunk 1 above)
Comment (sgraf812) [inline tests/semantic/compare_different_classes.invalid.mj]: Nope, it's not. See #23 and the respective [subparagraph 15.21.3](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.21.3) of 15.21 (the one you cited)

Comment (parttimenerd) [inline tests/semantic/compare_different_classes.invalid.mj]: You're correct. Thanks for your comment.

**Comment by moben** [comment]
done

**Pull request merged**